elide-core/src/main/java/com/yahoo/elide/resources/JsonApiEndpoint.java

```diff
@@ -112,13 +113,29 @@ public Response patch(
      */
     @DELETE
     @Path("{path:.*}")
+    @Consumes("application/vnd.api+json")
     public Response delete(
         @PathParam("path") String path,
         @Context SecurityContext securityContext,
         String jsonApiDocument) {
         return build(elide.delete(path, jsonApiDocument, getUser.apply(securityContext)));
     }
 
+    /**
+     * Delete resource handler (expects no body and no content-type header).
+     *
+     * @param path request path
+     * @param securityContext security context
+     * @return response
+     */
+    @DELETE
+    @Path("{path:.*}")
+    public Response deleteWithoutBody(
```

Comment:
@Koka I was thinking of changing the signatures to be:

```
@DELETE
@Path("{path:.*}")
public Response delete(
         @PathParam("path") String path,
          @Context SecurityContext securityContext)
``` and
```
@DELETE
@Path("{path:.*}")
@Consumes("application/vnd.api+json")
public Response deleteRelationship(
         @PathParam("path") String path,
          @Context SecurityContext securityContext,
          String jsonApiDocument)
```

Your fix is a minor change, changing the signatures like this a breaking change but is more explicit in the need for both versions. I'm not sure what our plans are for cutting more releases before 3.0. If you need #321 fixed before 3.0 drops then let's get this PR merged and clean things up in 3.0. Otherwise my vote is to do things right the first time.